nitime/timeseries.py

```diff
@@ -152,7 +152,7 @@ class instance, or an int64 array in the base unit of the module
                 e_s += 'TimeArray in object, or int64 times, in %s' % base_unit
                 raise ValueError(e_s)
 
-            time = np.array(data, copy=False)
+            time = np.asarray(data)
```

Comment:
Fixes
```
>           time = np.array(data, copy=False)
E           ValueError: Unable to avoid copy while creating an array as requested.
E           If using `np.array(obj, copy=False)` replace it with `np.asarray(obj)` to allow a copy when needed (no behavior change in NumPy 1.x).
E           For more details, see https://numpy.org/devdocs/numpy_2_0_migration_guide.html#adapting-to-changes-in-the-copy-keyword.
```

Reply:
If you look at the context, nitime was attempting to respect the `copy=False` flag of the calling function. I think it probably makes sense to be as strict as numpy here, and raise the exception. Then the test cases should be updated with new expectations about whether a copy is made or not.

Alternately, I suppose you could update the documentation to say that `copy=False` is a best-effort attempt.

Reply:
@arokem is the decider.

Reply:
I am inclined to go with the solution proposed here and add documentation. IIUC, the documentation will say that we are ignoring the request to set `copy=False`?

Reply:
Rather that it's best effort and will silently fall back to copying if no-copy isn't possible.